references/depth/stereo/utils/losses.py

```diff
@@ -0,0 +1,470 @@
+from typing import List, Optional
+
+import torch
+from torch import nn, Tensor
+from torch.nn import functional as F
+from torchvision.prototype.models.depth.stereo.raft_stereo import grid_sample, make_coords_grid
+
+
+def make_gaussian_kernel(kernel_size: int, sigma: float) -> torch.Tensor:
+    """Function to create a 2D Gaussian kernel."""
+
+    x = torch.arange(kernel_size, dtype=torch.float32)
+    y = torch.arange(kernel_size, dtype=torch.float32)
+    x = x - (kernel_size - 1) / 2
+    y = y - (kernel_size - 1) / 2
+    x, y = torch.meshgrid(x, y)
+    grid = (x**2 + y**2) / (2 * sigma**2)
+    kernel = torch.exp(-grid)
+    kernel = kernel / kernel.sum()
+    return kernel
+
+
+def _sequence_loss_fn(
+    flow_preds: List[Tensor],
+    flow_gt: Tensor,
+    valid_flow_mask: Tensor,
+    gamma: Tensor,
+    max_flow: int = 256,
+    exclude_large: bool = False,
+    weights: Optional[Tensor] = None,
+):
+    """Loss function defined over sequence of flow predictions"""
+    torch._assert(
+        gamma < 1,
+        "sequence_loss: `gamma` must be lower than 1, but got {}".format(gamma),
+    )
+
+    if exclude_large:
+        # exclude invalid pixels and extremely large diplacements
+        flow_norm = torch.sum(flow_gt**2, dim=1).sqrt()
+        valid_flow_mask = valid_flow_mask & (flow_norm < max_flow)
```

Comment:
`max_flow` would be enough if we assume users to use our transforms. If that assumption is correct than that term is not needed.

Reply:
Unfortunately, because some datasets have `None` returns for the disparity mask, whilst others don't. We need a transform that fills in that `None` with something. Even if `max_disparity` is not needed, we still have to impute somehow the `None`.

Reply:
Currently we allow `max_disparity` to be optional, and if it is `None` it will cause the problem you mention here. I think one possible solution is to let `valid_flow_mask` to be `Optional` as well here, and we treat `None` as if it is `True` everywhere. As indicated on previous comment, lets discuss on this particular topic first.